client/src/app/components/Layout/Layout.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Redirect, Route, NavLink, Switch, withRouter } from 'react-router-dom';
+
+import ShoppingList from 'modules/shopping-list';
+import Dashboard from 'modules/dashboard';
+import Cohort from 'modules/cohort';
+import AuthBox from 'modules/legacy/AuthBox';
+import Toolbar from '../Toolbar/Toolbar';
+import { setCurrentUser } from 'modules/legacy/mainActions';
+import { UserPropType } from 'common/constants/propTypes';
+
+class Layout extends Component {
+  componentDidMount() {
+    this.setAuthenticationState();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { currentUser, history } = this.props;
+    const {
+      location: { pathname }
+    } = prevProps;
+    if (!prevProps.currentUser && currentUser) {
```

Comment:
please destructure this from prev props to make it more readable and understandable